stream-chat-android-state/src/main/java/io/getstream/chat/android/offline/extensions/ChatClient.kt

```diff
@@ -289,6 +315,29 @@ public fun ChatClient.cancelEphemeralMessage(message: Message): Call<Boolean> {
     }
 }
 
+/**
+ * Attempts to fetch the message from offline cache before making an API call.
+ *
+ * @param messageId The id of the message we are fetching.
+ *
+ * @return The message with the corresponding iID wrapped inside a [Call].
+ */
+@InternalStreamChatApi
+@CheckResult
+public fun ChatClient.getMessageUsingCache(
+    messageId: String,
+): Call<Message> {
+    return CoroutineCall(state.scope) {
+        val message = logic.getMessageById(messageId) ?: logic.getMessageByIdFromDb(messageId)
```

Comment:
@TGazica is of the opinion that API calls should get priority here instead of cache/ DB calls so that we can have a better assurance that we are fetching latest data.

Input is appreciated.